SConstruct

```diff
@@ -1234,6 +1234,14 @@ def generateDocs( target, source, env ) :
 	# Run a script in the document source tree. These are used to
 	# autogenerate source files for processing by sphinx.
 
+	# SIP on MacOS prevents DYLD_LIBRARY_PATH being passed down so we make sure
+	# we pass through to gaffer the other base vars it uses to populate paths
+	# for third-party support
+
+	docEnv = dict( env["ENV"] )
+	for v in ( 'ARNOLD_ROOT', 'DELIGHT' ) :
+		docEnv[ v ] = os.environ.get( v, '' )
```

Comment:
We shouldn't be pulling from the external environment here; we should be getting the values from the build arguments instead. I think we may as well do this up around line 538 where we're setting up `commandEnv`, and then we can leave `generateDocs` unchanged. i.e. `commandEnv["ENV"]["ARNOLD_ROOT"] = commandEnv["ARNOLD_ROOT"]`.

Reply:
Ahh yes!

Reply:
https://github.com/GafferHQ/gaffer/commit/64718c7293a557bdb3684cff67c13a647a73bd48